packages/cardano-services/src/PgBoss/stakePoolBatchDelistHandler.ts

```diff
@@ -0,0 +1,42 @@
+import { CustomError } from 'ts-custom-error';
+import { MissingProgramOption } from '../Program/errors/MissingProgramOption';
+import { POOL_DELIST_SCHEDULE, WorkerHandlerFactory } from './types';
+import { PoolDelistedEntity } from '@cardano-sdk/projection-typeorm';
+import { createSmashStakePoolDelistedService } from '../StakePool/HttpStakePoolMetadata/SmashStakePoolDelistedService';
+
+const createService = (smashUrl: string | undefined) => {
+  if (!smashUrl) throw new MissingProgramOption(POOL_DELIST_SCHEDULE, 'smash-url');
+
+  return createSmashStakePoolDelistedService(smashUrl);
+};
+export const stakePoolBatchDelistHandlerFactory: WorkerHandlerFactory = (options) => {
+  const { logger, smashUrl, dataSource } = options;
+
+  const service = createService(smashUrl);
+  const repo = dataSource.getRepository(PoolDelistedEntity);
+
+  return async (_data: unknown) => {
+    logger.info(`Getting list of delisted pools from ${smashUrl}`);
+    const delists = await service.getDelistedStakePoolIds();
+
+    if (delists instanceof CustomError) {
```

Comment:
I like this pattern. Returning error instead of throwing it, sounds more granular, explicit.

Recently I red this [article](https://medium.com/javascript-in-plain-english/stop-using-try-catch-and-use-this-instead-2742b51cf97b) on Medium

Reply:
I have mixed feelings about this: on one hand errors are very common in JS, you have to build abstraction layers for everything to not have to worry that something would throw. So it's difficult to get away from, might as well embrace it :sweat_smile: 

On the other hand, when using TypeScript, returning errors has a clear advantage that it forces documenting the errors (as types) as well as handling them.

A couple other simple ways to do that:
```ts
// tuples are simple to implement, but you can technically return `[undefined, undefined]`
async getDelistedStakePoolIds(): Promise<[CustomError | undefined, Array<string> | undefined]>

// a little more preparation but more a more accurate representation of what we're trying to do
type Ok<R> = { ok: true; result: R };
type Err<E> = { ok: false; error: E };
type OkOrErr<T, E> = Ok<R> | Err<E>;
async getDelistedStakePoolIds(): Promise<OkOrErr<Array<string>, CustomError>>
```